src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,232 @@
+
+        // if entity is in colony -> remove from list
+        final List<Entity> entitiesToRemove = event.getAffectedEntities().stream()
+                                                .filter(getEntitiesInColony)
+                                                .collect(Collectors.toList());
+        event.getAffectedBlocks().removeAll(blocksToRemove);
+        event.getAffectedEntities().removeAll(entitiesToRemove);
+    }
+
+    /**
+     * ExplosionEvent.Start handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Start event)
+    {
+        if (colony.isCoordInColony(event.getWorld(), new BlockPos(event.getExplosion().getPosition())))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * PlayerInteractEvent handler.
+     * <p>
+     * Check, if a player right clicked a block.
+     * Deny if:
+     * - If the block is in colony
+     * - block is AbstractBlockHut
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+            // Huts
+            if (block instanceof AbstractBlockHut &&
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
+     * Check Permissions for Container Open Events.
+     *
+     * @param event the event to check on.
+     */
+    @SubscribeEvent
+    public void on(final PlayerContainerEvent.Open event)
+    {
+        if (this.colony.isCoordInColony(event.getEntity().getEntityWorld(), event.getEntity().getPosition())
+              && !this.colony.getPermissions().isColonyMember(event.getEntityPlayer()))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * ItemTossEvent handler.
+     *
+     * @param event ItemTossEvent
+     */
+    @SubscribeEvent
+    public void on(ItemTossEvent event)
+    {
+        final EntityPlayer playerIn = event.getPlayer();
+        if (colony.isCoordInColony(playerIn.getEntityWorld(), playerIn.getPosition()))
+        {
+            Permissions.Rank rank = colony.getPermissions().getRank(playerIn);
+
+            if (rank.ordinal() < Permissions.Rank.NEUTRAL.ordinal())
+            {
+                /*
+                    this will delete the item entirely:
+                    Canceling the event will stop the items from entering the world,
+                    but will not prevent them being removed from the inventory
+                    - and thus removed from the system.
+                 */
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)